mlrun/api/db/base.py

```diff
@@ -215,7 +215,9 @@ def delete_function(self, session, project: str, name: str):
         pass
 
     @abstractmethod
-    def list_functions(self, session, name=None, project="", tag="", labels=None):
+    def list_functions(
+        self, session, name=None, project="", tag="", labels=None, hash_key=None
+    ):
```

Comment:
lets add type hinting especially for the attribute u added and if possible for the other as well